internal/testrunner/runners/system/servicedeployer/kubernetes.go

```diff
@@ -0,0 +1,269 @@
+	}
+	return string(containerIDs[0]), nil
+}
+
+func connectControlPlaneToElasticStackNetwork(controlPlaneContainerID string) error {
+	stackNetwork := fmt.Sprintf("%s_default", stack.DockerComposeProjectName)
+
+	logger.Debugf("check network connectivity between service container %s (ID: %s) and the stack network %s", kindControlPlaneContainerName, controlPlaneContainerID, stackNetwork)
+	cmd := exec.Command("docker", "network", "inspect", stackNetwork)
+	errOutput := new(bytes.Buffer)
+	cmd.Stderr = errOutput
+	output, err := cmd.Output()
+	if err != nil {
+		return errors.Wrapf(err, "could not inspect the stack network (stderr=%q)", errOutput.String())
+	}
+
+	var networkDescriptions []struct {
+		Containers map[string]struct {
+			Name string
+		}
+	}
+	err = json.Unmarshal(output, &networkDescriptions)
+	if err != nil {
+		return errors.Wrapf(err, "can't unmarshal network inspect for %s (stderr=%q)", stackNetwork, errOutput.String())
+	}
+
+	if len(networkDescriptions) != 1 {
+		return fmt.Errorf("expect single network inspect record, got %d entries", len(networkDescriptions))
+	}
+
+	for _, c := range networkDescriptions[0].Containers {
+		if c.Name == kindControlPlaneContainerName {
+			logger.Debugf("container %s is already attached to the %s network", kindControlPlaneContainerName, stackNetwork)
+			return nil
+		}
+	}
+
+	logger.Debugf("attach service container %s (ID: %s) to stack network %s", kindControlPlaneContainerName, controlPlaneContainerID, stackNetwork)
+	cmd = exec.Command("docker", "network", "connect", stackNetwork, controlPlaneContainerID)
+	errOutput = new(bytes.Buffer)
+	cmd.Stderr = errOutput
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "could not attach \"%s\" container to the stack network (stderr=%q)", kindControlPlaneContainerName, errOutput.String())
+	}
+	return nil
+}
+
+func installElasticAgentInCluster() error {
+	logger.Debug("install Elastic Agent in the Kubernetes cluster")
+
+	elasticAgentFile, err := install.KubernetesDeployerElasticAgentFile()
+	if err != nil {
+		return errors.Wrap(err, "can't locate Kubernetes file for Elastic Agent in ")
+	}
+
+	err = modifyKubernetesResources(true, elasticAgentFile)
+	if err != nil {
+		return errors.Wrap(err, "can't install Elastic-Agent in Kubernetes cluster")
+	}
+	return nil
+}
+
+func modifyKubernetesResources(apply bool, definitionPaths ...string) error {
+	var args []string
+	if apply {
+		args = append(args, "apply")
+	} else {
+		args = append(args, "delete")
+	}
+
+	for _, definitionPath := range definitionPaths {
+		args = append(args, "-f")
+		args = append(args, definitionPath)
+	}
+
+	cmd := exec.Command("kubectl", args...)
+	errOutput := new(bytes.Buffer)
+	cmd.Stderr = errOutput
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "kubectl apply failed (stderr=%q)", errOutput.String())
+	}
+	return nil
```

Comment:
We are doing something similar in `verifyKindContext`. Does it make sense to extract this part into its own function  for running generic `kubectl` commands?

Reply:
Maybe these two functions are not good candidates to get unified, but this might a good point to extract it to the `kind` package.

Reply:
++ to `kind` package. That would be even better!

Reply:
Fixed. Introduced `kind`, `kubectl` and added some changes to `docker` package.